pkg/controller/pagerdutyintegration/pagerdutyintegration_controller_test.go

```diff
@@ -243,38 +248,52 @@ func TestReconcilePagerDutyIntegration(t *testing.T) {
 		{
 			name: "Test Managed, No Finalizer, Not Deleting, PD Not Setup",
 			localObjects: []runtime.Object{
-				testClusterDeployment(true, true, false, false),
+				testClusterDeployment(true, true, false, false, false),
 				testPDISecret(),
 				testPagerDutyIntegration(),
 			},
 			expectPDSetup: true,
 			setupPDMock: func(r *mockpd.MockClientMockRecorder) {
-				r.CreateService(gomock.Any()).Return(testIntegrationID, nil).Times(1)
+				r.CreateService(gomock.Any()).Return(testIntegrationID, nil).Times(1).DoAndReturn(
+					func(data *pd.Data) (string, error) {
+						data.ServiceID = "XYZ123"
+						data.IntegrationID = "LMN456"
+						return data.IntegrationID, nil
+					})
 				r.GetIntegrationKey(gomock.Any()).Return(testIntegrationID, nil).Times(1)
 				r.DeleteService(gomock.Any()).Return(nil).Times(0)
+				r.DisableService(gomock.Any()).Return(nil).Times(0)
+				r.EnableService(gomock.Any()).Return(nil).Times(0)
 			},
 		},
 		{
 			name: "Test Managed, Finalizer, Not Deleting, PD Not Setup",
 			localObjects: []runtime.Object{
-				testClusterDeployment(true, true, true, false),
+				testClusterDeployment(true, true, true, false, false),
 				testPDISecret(),
 				testPagerDutyIntegration(),
 			},
 			expectPDSetup: true,
 			setupPDMock: func(r *mockpd.MockClientMockRecorder) {
-				r.CreateService(gomock.Any()).Return(testIntegrationID, nil).Times(1)
+				r.CreateService(gomock.Any()).Return(testIntegrationID, nil).Times(1).DoAndReturn(
```

Comment:
See comment https://github.com/openshift/pagerduty-operator/pull/157/files#r690849779 for these changes